Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -5523,8 +5523,12 @@ ITransportChannel transportChannel
             EndpointDescription endpoint = m_endpoint.Description;
             SignatureData clientSignature = SecurityPolicies.Sign(m_instanceCertificate, endpoint.SecurityPolicyUri, dataToSign);
 
-            // check that the user identity is supported by the endpoint.
-            UserTokenPolicy identityPolicy = endpoint.FindUserTokenPolicy(m_identity.TokenType, m_identity.IssuedTokenType);
+            UserTokenPolicy identityPolicy = m_endpoint.Description.FindUserTokenPolicy(m_identity.PolicyId);
+            if (identityPolicy == null)
```

Comment:
@KircMax Does the second "FindUserTokenPolicy(m_identity.TokenType, m_identity.IssuedTokenType)" make any difference (is it ever going to get executed) ? If its purpose is to take into account a PolicyId change on the server side during the disconnect time, I still cannot see a renewal of the m_endpoint.Description being made in the reconnect scenario path...

Reply:
I was not really sure... 
I am not too familiar with the codebase and wanted to keep the changes 'defensive' but my feeling was it should never be necessary...
I'd also not really expect a server that had PolicyId changes during the disconnect time be able to reactivate a session that has been before that change, therefore I think we could also get rid of that second "FindUserTokenPolicy(m_identity.TokenType, m_identity.IssuedTokenType)"

Reply:
--> I'll remove the second one and push the update.